lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -728,17 +730,19 @@ SPIRVValue *LLVMToSPIRV::transConstant(Value *V) {
 }
 
 SPIRVValue *LLVMToSPIRV::transValue(Value *V, SPIRVBasicBlock *BB,
-                                    bool CreateForward) {
+                                    bool CreateForward,
+                                    FuncTransMode FuncTrans) {
   LLVMToSPIRVValueMap::iterator Loc = ValueMap.find(V);
-  if (Loc != ValueMap.end() && (!Loc->second->isForward() || CreateForward))
+  if (Loc != ValueMap.end() && (!Loc->second->isForward() || CreateForward) &&
+      !(FuncTrans == FuncTransMode::Pointer && isa<Function>(V)))
```

Comment:
Am I right that the purpose of this "if" is to avoid returning forward-declarations of functions when `OpConstFunctionPointerINTEL` is needed to be generated?

Reply:
Yep

Reply:
Done